Sources/GRPCCore/Call/Server/Internal/ServerRPCExecutor.swift

```diff
@@ -119,43 +119,29 @@ struct ServerRPCExecutor {
       _ context: ServerContext
     ) async throws -> StreamingServerResponse<Output>
   ) async {
-    await withTaskGroup(of: ServerExecutorTask.self) { group in
+    await withTaskGroup(of: Void.self) { group in
       group.addTask {
-        let result = await Result {
+        do {
           try await Task.sleep(for: timeout, clock: .continuous)
+          context.cancellation.cancel()
+        } catch {
+          ()  // Only cancel the RPC if the timeout completes.
```

Comment:
Hmm, not sure about this one. My reason for not doing it is that task cancellation is more forceful than RPC cancellation (which is really just a suggestion) so if the parent task is cancelled then the task running the RPC will also be cancelled. I'm not sure if it sense to also signal RPC cancellation because the user won't be able to do anything about it.

Reply:
I see, yeah okay that makes sense. For a moment I thought that the RPC would continue to run but you're right, it will also be cancelled, so I don't think we should signal RPC cancellation in that case.